cmd/compareBedDistanceBasedOnName/compareBedDistanceBasedOnName.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+func compareBedDistanceBasedOnName(inputBed string, genomeBed string, outBed string) {
+	out := fileio.EasyCreate(outBed)
+	input := bed.Read(inputBed)
+	genome := bed.Read(genomeBed)
+
+	genomeMap := make(map[string][]bed.Bed)
+	var i int
+
+	for i = range genome {
+		genomeMap[genome[i].Name] = append(genomeMap[genome[i].Name], genome[i])
+	}
+
+	var inputNameField []bed.Bed
+	var found bool
+	var distance int
+	var lowestDistance int
+	var err error
+	var j int
+
+	for i := 0; i < len(input); i++ {
+		lowestDistance = 0
+		inputNameField, found = genomeMap[input[i].Name]
+		if found != true {
+			log.Fatalf("Did not find %s", input[i].Name)
+		}
+		for j = range inputNameField {
+			distance, err = bed.CompareDistance(inputNameField[j], input[i])
+			if j == 0 {
+				lowestDistance = distance
+			} else {
+				if distance < lowestDistance {
+					lowestDistance = distance
+				}
+			}
+		}
+
+		if err != nil {
+			log.Fatalf("Unable to compare distance, error message: %s", err)
+		}
+		input[i].Score = lowestDistance
+		if input[i].FieldsInitialized < 5 {
+			input[i].FieldsInitialized = 5
+		}
+		bed.WriteToFileHandle(out, input[i])
+	}
+	err = out.Close()
+	if err != nil {
```

Comment:
Usually we call exception.PanicOnErr(err) here, which automatically checks if the err is not-nil

Reply:
does that still fatal or just report a message stating there was a problem? I actually dont want the program to keep going if piped into something downstream as I could have missing data if the file didnt close.

Reply:
updated, thanks!